### PR TITLE
Collapse Shuffle of strided-loads to shuffle's of non-strided loads

### PR DESCRIPTION
**Pull request by aankit-ca**
@dsharletg I don't think we will access any out of bounds memory with the change. Can you take a look once? If you think this is a good change then I'll add the comments and testcase and clean the patch.

### CODE REVIEW

File: src/Simplify_Shuffle.cpp [hunk 1]
```diff
@@ -282,10 +282,36 @@ Expr Simplify::visit(const Shuffle *op, ExprInfo *bounds) {
...
+                const int64_t *stride = as_const_int(ramp->stride);
+                if (stride && (*stride != 1) && is_const_one(load->predicate)) {
+                    changed = true;
+                    for (size_t j = 0; j < new_indices.size(); j++) {
+                        int index = new_indices[j];
+                        if ((index >= curr_min) && (index < curr_max)) {
+                            new_indices[j] = index * (*stride);
+                        }
+                    }
+                    Expr new_load_index = Ramp::make(ramp->base, 1, ramp->lanes * (*stride));
```
Comment (abadams) [inline src/Simplify_Shuffle.cpp:301]: Consider ramp->base = 0, ramp->lanes = 2, ramp->stride = 10

The original load accesses two elements at 0 and 10. This new load accesses all elements from 0 to 19 inclusive.

Isn't that a potentially-new out of bounds load?

Comment (aankit-ca) [reply src/Simplify_Shuffle.cpp]: Aah. Got it. I think this expression would prevent this from happening.

`Expr new_load_index = Ramp::make(ramp->base, 1, (ramp->lanes - 1) * (*stride) + 1);`

**Comment by dsharletg** [comment]
I think this might be a harmful change on targets that can do strided loads well, like ARM?

There probably also needs to be some upper bound on the stride that we do this with. Rewriting a load with a large stride into a dense stride might also be harmful.

**Comment by aankit-ca** [comment]
Ok. Closing this for now. Dillon suggested a better way to do this in a way that does not affect other targets. Will try that in the future.

**Pull request closed without merging**